dotnet/src/Agents/Abstractions/Extensions/ChatHistoryExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Microsoft.SemanticKernel.Agents.Extensions;
+
+/// <summary>
+/// Extension methods for <see cref="ChatMessageContent"/>
+/// </summary>
+internal static class ChatHistoryExtensions
+{
+    /// <summary>
+    /// Enumeration of chat-history in descending order.
+    /// </summary>
+    /// <param name="history">The chat-history</param>
+    public static IEnumerable<ChatMessageContent> ToDescending(this ChatHistory history)
```

Comment:
Why are we doing this?

If the order of messages is important we should look at the design for this e.g. Roger suggested we should add a timestamp to chat messages. I'd like to understand do we have a good use case to justify this change.